Proposed/asc058.md

```diff
@@ -0,0 +1,40 @@
+#### **Title**
+Add `attach_sourcing_info` as a datatype for the `informatics` instruction attribute
+
+#### **Authorship**
+Al Colmenar <al.colmenar@strateos.com>
+
+#### **Motivation**
+Currently, certain instructions operate on resources that don't contain the information needed to further drill down into the sourcing information associated to a particular container. This additional sourcing information is required when a customer wants to provide the vendor and supplier of the resource to be used in a run. An example of an instruction where this would be useful is the `provision` Autoprotocol instruction which is meant to describe a task of provisioning a container for a given resource.
+
+#### **Proposal**
+A new data type for the optional `informatics` attribute on instructions utilizing it will be added. This data type will allow a customer to define the sourcing information for a resource used in a run. Also it will assist the system to properly filter the list of kits associated to a given resource for instructions utilizing resources, e.g. the provision `op`, reducing the list of eligible containers shown when an operator selects the container to use.
+
+Currently, the optional `informatics` attribute only supports the data type, `attach_compounds`. We are proposing to add `attach_sourcing_info` as a data type. The `data` field within the object for the `informatics` attribute will have a format specific to `attach_sourcing_info` containing new string attributes: `vendor` and `supplier`.
+
+```
+"instructions": [
+  {
+    "op": String
+    ...,
+    /*
+     * optional parameter describing a list of intended effects or changes
+     * for this instruction.
+     */
+    "informatics": [
+      {
+        /* New value for type */
+        "type": Enum("attach_sourcing_info"),
+        /* Details of the type. */
+        "data": {
+          "supplier": String,
+          "vendor": String
```

Comment:
Could you elaborate on why this information is needed at the autoprotocol level and can't be inferred from resources used at Strateos?

Reply:
I addressed this slightly in the other comment, but the main reason is that the resource model doesn't first class supplier and vendor and the same generic resource can be apart of multiple kits which currently hold the supplier and vendor information. Prior to run submission, an operator has to select a container with the resource and shouldn't select any container if the user specified sourcing information.

Reply:
then let's update the resource model!

if we're specifying vendor in autoprotocol, and the resource model doesnt support it, how can we look up the resource by vendor in web?